packages/fx-core/src/component/deploy/azureDeployDriver.ts

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DeployStepArgs,
+  AzureResourceInfo,
+  DeployTypes,
+  DeployContext,
+  AzureUploadConfig,
+  AxiosOnlyStatusResult,
+} from "./deployTypes";
+import { checkMissingDeployArgs } from "../utils/common";
+import {
+  DeployExternalApiCallError,
+  DeployPrerequisiteError,
+  DeployTimeoutError,
+} from "../error/deployError";
+import { AzureAccountProvider, LogProvider } from "@microsoft/teamsfx-api";
+import { TokenCredentialsBase } from "@azure/ms-rest-nodeauth";
+import { BaseDeployDriver } from "./deployDriver";
+import { Base64 } from "js-base64";
+import * as appService from "@azure/arm-appservice";
+import { DeployConstant } from "../constant/deployConstant";
+import { default as axios } from "axios";
+import { AxiosZipDeployResult } from "./deployTypes";
+import { waitSeconds } from "../../common";
+
+export abstract class AzureDeployDriver extends BaseDeployDriver {
```

Comment:
Overall the code looks clear and easy to understand, and I really appreciate your effort on the code/design quality.